lib/route_consistency_checker.rb

```diff
@@ -0,0 +1,145 @@
+require "set"
+
+require "gds_api/router"
+
+class RouteConsistencyChecker
+  attr_reader :errors
+
+  def initialize(routes, router_data)
```

Comment:
on longer discussion - we thought about making it so this only accepts data and that we have a specific class that converts the CSV into open structs